tests/__init__.py

```diff
@@ -359,3 +361,58 @@ def assert_url_equal(url1, url2):
     assert_equal(parts1.hostname, parts2.hostname)
     assert_equal(parts1.port, parts2.port)
     assert_equal(parse_qs(parts1.query), parse_qs(parts2.query))
+
+
+class HTTPStubberException(Exception):
+    pass
+
+
+class RawResponse(BytesIO):
+    def stream(self, **kwargs):
+        contents = self.read()
+        while contents:
+            yield contents
+            contents = self.read()
+
+
+class ClientHTTPStubber(object):
```

Comment:
Wait, why is this in `tests` and not `stub`? Customers won't actually be able to use it here. Also, docs / examples should be added.

Reply:
We did not want to make it a public interface quite yet.

Reply:
This PR is really for the `before-send` event. This class is currently something we're using to exercise the event and ensure that it works for what we need it to do. Making this HTTP stubber an actual public stubber is a feature we can come back to later.